aip/cloud/2510.md

```diff
@@ -67,6 +67,46 @@ considered PII and user data, but project numbers are not.
 Therefore, when an internal service calls an external Google APIs, it
 **should** use project numbers for making API requests.
 
+### Resource References
+
+Project IDs also appear in [resource names][], which are used
+when referring to a separate resource ([example][]).
+
+In these situations, the value of the resource reference should be preserved
+regardless if the user has sent a project ID or number to identify the main
+resource.
+
+For example, for an existing resource book:
+
+```proto
+book {
+  // project-id: "foo"
+  // project-number: 12345
+  name: "project/12345/books/game-of-tones"
+  shelf: "project/123456/shelves/top-row"
+}
+```
+
+A request for the book in the format of `projects/foo/books/game-of-tones`
+should return:
+
+```proto
+book {
+  // Returning the project ID, since that was what was sent.
+  name: "project/foo/books/game-of-tones"
```

Comment:
So in the essence you want services to return this string unchanged, correct?

Reply:
this particular line refers to the resource name, which should match the request (project-id or project-number).

The new section does state that resource name in other contexts (e.g. as a resource name) should be unchanged.